atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2583,7 +2567,11 @@ private void handleKeyAlreadyExistsException(
                                 .immutableTimestampLock()
                                 .map(token -> token.toSafeArg("immutableTimestampLock"))
                                 .orElseGet(() -> SafeArg.of("immutableTimestampLock", null)),
-                        lockCheckResult.userProvidedLock().toSafeArg("commitLocksToken"));
+                        SafeArg.of(
+                                "allLockTokens",
+                                lockCheckResult.allLockTokens().stream()
+                                        .map(LockToken::getRequestId)
+                                        .collect(Collectors.toSet())));
```

Comment:
The conflict checking group: these usages are semantics-preserving, since they happen after `commitWrites` where we acquire the lock and register it.

I guess there is a **compile break** (and some cursory searches on internal code search tool also did not find anything), though the protected visibility of some of these methods is a little concerning given we are relying quite a bit on the `commitWrites` implementation. I can't think of a legitimate use case for calling directly into these, and wouldn't block any explicit action on this, though I do wonder why they are protected / if they _can_ be made private that would be nice.

Reply:
You mean the #allLockTokens is changed now? This is strictly private API, no?

Reply:
Or TransactionLockManager#getExpiredImmutableTimestampAndCommitLocksWithFullSummary? handleKeyAlreadyExistsException and putCommitTimestamp are all private as well